lib/manageiq/release/code_climate.rb

```diff
@@ -0,0 +1,176 @@
+require 'rest-client'
+require 'json'
+require 'more_core_extensions/core_ext/array'
+require 'more_core_extensions/core_ext/hash'
+
+module ManageIQ
+  module Release
+    class CodeClimate
+      def self.api_token
+        @api_token ||= ENV["CODECLIMATE_API_TOKEN"]
+      end
+
+      def self.api_token=(token)
+        @api_token = token
+      end
+
+      def self.badge_name
+        "Maintainability"
+      end
+
+      def self.badge_details(repo, token)
+        {
+          "description" => badge_name,
+          "image"       => "https://api.codeclimate.com/v1/badges/#{token}/maintainability",
+          "url"         => "https://codeclimate.com/github/#{repo.github_repo}/maintainability"
+        }
+      end
+
+      def self.coverage_badge_name
+        "Test Coverage"
+      end
+
+      def self.coverage_badge_details(repo, token)
+        {
+          "description" => coverage_badge_name,
+          "image"       => "https://api.codeclimate.com/v1/badges/#{token}/test_coverage",
+          "url"         => "https://codeclimate.com/github/#{repo.github_repo}/test_coverage"
+        }
+      end
+
+      attr_reader :repo, :dry_run
+
+      def initialize(repo, dry_run: false)
+        @repo    = repo
+        @dry_run = dry_run
+      end
+
+      def save!
+        write_codeclimate_yaml
+        write_rubocop_yamls
+      end
+
+      def enable
+        ensure_enabled
+      end
+
+      def badge_token
+        ensure_enabled
+        @response.fetch_path("data", 0, "attributes", "badge_token")
+      end
+
+      def badge_details
+        self.class.badge_details(repo, badge_token)
+      end
+
+      def coverage_badge_details
+        self.class.coverage_badge_details(repo, badge_token)
+      end
+
+      def test_reporter_id
+        ensure_enabled
+        @response.fetch_path("data", 0, "attributes", "test_reporter_id")
+      end
+
+      def set_travis_test_reporter_id
+        Travis.new(repo, dry_run: dry_run).set_env("CC_TEST_REPORTER_ID" => test_reporter_id)
+      end
+
+      private
+
+      def ensure_enabled
+        return if @enabled
+
+        @response =
+          if dry_run
+            puts "** dry-run: RestClient.get(\"https://api.codeclimate.com/v1/repos?github_slug=#{repo.github_repo}\", #{headers})"
+            {"data" => [{"attributes" => {"badge_token" => "0123456789abdef01234", "test_reporter_id" => "0123456789abcedef0123456789abcedef0123456789abcedef0123456789abc"}}]}
+          else
+            JSON.parse(RestClient.get("https://api.codeclimate.com/v1/repos?github_slug=#{repo.github_repo}", headers))
```

Comment:
Answering your [question from the previous review](https://github.com/ManageIQ/manageiq-release/pull/80/files#r255660275):

> Is there a nice way to do get with headers?

Yes, I agree that it is a pain, but here is a way that you could do `GET` with a few lines of code:

```ruby
require 'uri'
require 'json'
require 'openssl'
require 'net/http'

private

def get path, params = {}
	route = path.dup
	unless params.empty?
		route << "?".freeze
		route << URI.encode_www_form(params)
	end
  # using the existing `headers` method you already have here, but you could
  # pass it in as a param if you really wanted
	request = Net::HTTP::Get.new route, headers

  http_send(request).body
end

# Just a wrapper around the existing `Net::HTTP.post`.  Could probably use the
# `http_send` method below, but I don't have a quick way of validating that at
# the moment.
def post path, payload, headers = nil
  Net::HTTP.post uri, payload, headers
end

CODE_CLIMATE_URL = URI.parse "https://api.codeclimate.com"

# Initialize Net::HTTP instance pointing to Code Climate
def http_send request_to_send
	Net::HTTP.new(CODE_CLIMATE_URL.host, CODE_CLIMATE_URL.port).tap do |http|
		http.use_ssl     = true
		http.verify_mode = OpenSSL::SSL::VERIFY_PEER
	end.start do |http|
		http.request request_to_send
	end
end


# Usage
#
#   get  '/v1/repos', 'github_slug' => repo.github_repo
#   post '/v1/github/repos', payload, headers
#
#   JSON.parse(get '/v1/repos', 'github_slug' => repo.github_repo)
```


(note, pulled from something else I am working on, so sorry if I forget to change a variable here to work with what you have)

Reply:
I did find that I don't need to do any SSL dance if I give an https URL, but the headers is gross.

Reply:
So, hmmm...that's a lot of code that is otherwise just

`RestClient.get(url, headers)`

Out of curiosity, what is the problem with rest-client?  I get that it's old, but it works, so I'm not sure why we wouldn't use it, particularly for non-deployable tools?